README.rst

```diff
@@ -21,6 +21,13 @@ a `tax-deductible donation`_ to help the project pay for developer time, profess
   :target: https://numfocus.org/project/mdanalysis
   :align: center
   :alt: NumFOCUS
+
```

Comment:
We don't need our logo here.

If anything, a smaller NumFOCUS logo would be nice. I am not 100% sure if the size parameter works with the rst rendered on GitHub, but that's something worthwhile trying.